### PR TITLE
websocket update

### PR DESCRIPTION
**Pull request by MilosKozak**

### CODE REVIEW

File: lib/websocket.js [hunk 1]
```diff
@@ -117,22 +112,83 @@ function init (env, ctx, server) {
...
+          return;
+        }
+        
+        if (data.collection === 'treatments' && !('eventType' in data.data)) {
+          data.data.eventType = '<none>';
+        }
+        if (!('created_at' in data.data)) {
+          data.data.created_at = new Date().toISOString();
+        }
+
```
Comment (jasoncalabrese) [inline lib/websocket.js]: Codacy Issue found: ["result" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1725154834/issues/source?bid=2452379&fileBranchId=3052457#l183)

File: lib/websocket.js [hunk 2]
```diff
@@ -3,17 +3,32 @@
 var levels = require('./levels');
 var times = require('./times');
 var calcData = require('./data/calcdelta');
-
+var _ = require('lodash');
```
Comment (jasoncalabrese) [inline lib/websocket.js]: Codacy Issue found: ["_" is defined but never used](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1725154834/issues/source?bid=2452379&fileBranchId=3052457#l6)

File: lib/data/ddata.js [hunk 3]
```diff
@@ -47,11 +51,11 @@ function init( ) {
     partition('treatments');
     partition('devicestatus');
 
-    result.first.sgvs = ddata.sgvs;
-    result.first.cals = ddata.cals;
+    result.first.sgvs = ddata.sgvs.filter(filterMax);
+    result.first.cals = ddata.cals.filter(filterMax);
```
Comment (jasoncalabrese) [inline lib/data/ddata.js]: Probably shouldn't filter cals, there won't be many and if we don't have 1 we can't show any raw

File: lib/websocket.js [hunk 4]
```diff
@@ -3,17 +3,31 @@
...
+       
 function init (env, ctx, server) {
 
   function websocket ( ) {
     return websocket;
   }
 
+  var log_yellow = '\x1B[33m';
+  var log_reset = '\x1B[0m';
+  var LOG_IO = log_yellow + 'IO: ' + log_reset;
```
Comment (jasoncalabrese) [inline lib/websocket.js]: fancy log colors :)

File: lib/data/ddata.js [hunk 5]
```diff
@@ -28,18 +28,22 @@ function init( ) {
...
       , rest: { }
     };
-
+    
     function recent (item) {
       return item.mills >= time - cutoff;
     }
 
+    function filterMax(item) {
+      return item.mills >= time - max
```
Comment (jasoncalabrese) [inline lib/data/ddata.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1725345901/issues/source?bid=2452379&fileBranchId=3052457#l42)

File: lib/data/ddata.js (hunk 3 above)
Comment (MilosKozak) [inline lib/data/ddata.js]: ok, removed

File: lib/client/renderer.js [hunk 6]
```diff
@@ -559,6 +559,17 @@ function init (client, d3) {
...
+  renderer.drawTreatments = function drawTreatment(client) {
+    // add treatment bubbles
+    client.ddata.treatments
+      .forEach(function (d) {
+        renderer.drawTreatment(d, {
+          scale: renderer.bubbleScale()
+          , showLabels: true
+        });
+      });
+  }
```
Comment (jasoncalabrese) [inline lib/client/renderer.js]: Codacy Issue found: [Missing semicolon.](https://www.codacy.com/app/Nightscout/cgm-remote-monitor/file/1736384266/issues/source?bid=2452379&fileBranchId=3052457#l571)

File: lib/websocket.js (hunk 4 above)
Comment (jasoncalabrese) [inline lib/websocket.js]: thinking about this a little more, instead of IO we should use WS, and maybe green, since yellow is usually a warning

**Pull request merged**